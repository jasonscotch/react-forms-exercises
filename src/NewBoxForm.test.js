import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

it("renders without crashing", function () {
    render(<NewBoxForm />);
    });
    
it("matches snapshot", function () {
    const { asFragment } = render(<NewBoxForm />);
    expect(asFragment()).toMatchSnapshot();
    });

it('adds a new box on form submission and resets the form', () => {
    // Define a mock addBox function
    const mockAddBox = jest.fn();

    const { getByLabelText, queryByText } = render(
    <NewBoxForm addBox={mockAddBox} />
    );

    const widthInput = getByLabelText('Width:');
    const heightInput = getByLabelText('Height:');
    const bgColorInput = getByLabelText('Background Color:');
    const submitButton = queryByText('Add Box');

    fireEvent.change(widthInput, { target: { value: '250px' } });
    fireEvent.change(heightInput, { target: { value: '250px' } });
    fireEvent.change(bgColorInput, { target: { value: 'teal' } });
    fireEvent.click(submitButton);

    expect(mockAddBox).toHaveBeenCalledWith({
    width: '250px',
    height: '250px',
    backgroundColor: 'teal',
    });

    expect(widthInput).toHaveValue(''); 
    expect(heightInput).toHaveValue(''); 
    expect(bgColorInput).toHaveValue('');
});
