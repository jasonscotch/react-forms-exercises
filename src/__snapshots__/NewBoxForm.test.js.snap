// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="width"
    >
      Width:
    </label>
    <input
      id="width"
      name="width"
      placeholder="Box Width"
      type="text"
      value=""
    />
    <label
      for="height"
    >
      Height:
    </label>
    <input
      id="height"
      name="height"
      placeholder="Box Height"
      type="text"
      value=""
    />
    <label
      for="backgroundColor"
    >
      Background Color:
    </label>
    <input
      id="backgroundColor"
      name="backgroundColor"
      placeholder="Box Background Color"
      type="text"
      value=""
    />
    <button>
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
